### PR TITLE
Add AssertJ Guava assertions guide

### PR DESCRIPTION
**Pull request by scordio**
Open points:
- [x] the javadoc URL should be parameterized with the `assertj-guava` version number

### CODE REVIEW

File: src/docs/asciidoc/user-guide/assertj-guava-assertions-guide.adoc [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+|{assertj-guava-javadoc-root}org/assertj/guava/api/OptionalAssert.html#isAbsent()[`isAbsent`]
+|Verifies that the actual `Optional` contained instance is absent/null.
+
+|{assertj-guava-javadoc-root}org/assertj/guava/api/OptionalAssert.html#isPresent()[`isPresent`]
+|Verifies that the actual `Optional` contains a (non-null) instance.
+|===
+
+[%autowidth]
+|===
+|--FIXME-- |Description
```
Comment (scordio) [inline src/docs/asciidoc/user-guide/assertj-guava-assertions-guide.adoc:89]: @joel-costigliola I was thinking about separating normal assertion methods from "extracting" / "navigational" methods. Any good naming for them?

Comment (joel-costigliola) [reply src/docs/asciidoc/user-guide/assertj-guava-assertions-guide.adoc:89]: Maybe "advanced usage" or "asserting on the contained value" ? 
the first is generic and not too descriptive, I kinda prefer the second but no strong opinion here

Comment (scordio) [reply src/docs/asciidoc/user-guide/assertj-guava-assertions-guide.adoc:89]: I'm looking for something that could also cover the various `extracting`, `first`, `last`, `asInstanceOf`, etc.

Comment (scordio) [reply src/docs/asciidoc/user-guide/assertj-guava-assertions-guide.adoc:89]: I'd put them together in a single table for the time being, going forward with the rest of the docs maybe a good name will pop up.

**Comment by joel-costigliola** [comment]
Looks good to me but hard to review without generating the doc, let's see IRL ;-)
We can always change it later if we want to.

**Pull request merged**